app/Model/CoEnrollmentFlow.php

```diff
@@ -847,85 +847,87 @@ public function configuredSteps($id) {
   /**
    * Duplicate an existing Enrollment Flow.
    *
-   * @since  COmanage Registry v0.9.2
+   * @since  COmanage Registry v4.5.0
    * @param  Integer $id CO Enrollment Flow ID
    * @return Boolean True on success
    * @throws InvalidArgumentException
    * @throws RuntimeException
    */
-  
+
   public function duplicate($id) {
-    // First pull all the stuff we'll need to copy.
-    
-    $args = array();
-    $args['conditions']['CoEnrollmentFlow.id'] = $id;
-    $args['contain']['CoEnrollmentAttribute'][] = 'CoEnrollmentAttributeDefault';
-    $args['contain'][] = 'CoEnrollmentSource';
-    
-    // This find will not pull archived or deleted attributes (as managed via
-    // Changelog behavior), which seems about right. However, we'll want to clear
-    // the attribute changelog metadata, below.
-    $ef = $this->find('first', $args);
-    
-    if(empty($ef)) {
+    // This maintains a per-Model map of old_id -> new_id
+    $idmap = array();
+
+    // (0) Start a transaction. If we error out, we can roll back.
+    $this->_begin();
+
+    // (1) First copy the CO itself
```

Comment:
We're copying the Enrollment Flow here, not the CO?

Reply:
done